usr/src/uts/common/fs/zfs/vdev_file.c

```diff
@@ -59,6 +59,11 @@ vdev_file_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
 	int error;
 
 	/*
+	 * Rotational optimizations only make sense on block devices
```

Comment:
nit: don't assume people have common sense. A better approach is:
Rotational optimizations only apply to block devices.

Reply:
... which means that the real value should be "rotational" rather than "non-rotational" thereby eliminating later confusion as new vdev types are added.
Indeed, in Linux, the block device queue sysfs parameter is "rotational"

Reply:
So under Linux in user space this tuning does appear as `/sys/block/<blkdev>/queue/rotational`, sadly on the kernel side the helper function is called `blk_queue_nonrot()`.  Which is why this ended up being named `vdev_nonrot`. I wouldn't mind renaming it to `vdev_rotational`.


While it's true that this entry is called "rotational" when users see this parameter listed in

Reply:
and the device property we added is "device-rotational" as well.

Reply:
I don't think we need a code change here, there is lots of existing baggage to deal with. For example, in illumos there are two flags in sd: un_f_is_rotational and un_f_is_solid_state.

I'd be happier with the comments taking the perspective that rotation is the exception that applies to block devices.

Reply:
@richardelling yes, I added the ```un_f_is_rotational``` exactly for this change as it's different from ```un_f_is_solid_state```.